MINOR: Upgrade gradle plugins and test libraries for Java 14 support

ijuma opened the pull request:
Also:
* Remove deprecated `=` in resolutionStrategy.
* Replace `AES/GCM/PKCS5Padding` with `AES/GCM/NoPadding`
in `PasswordEncoderTest`. The former is invalid and JDK 14 rejects it,
see https://bugs.openjdk.java.net/browse/JDK-8229043.

With these changes, the build works with Java 14 and Scala 2.12. The
same will apply to Scala 2.13 when Scala 2.13.2 is released (should
happen within 1-2 weeks).

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

ijuma committed:
MINOR: Upgrade plugins and test libraries for Java 14 support

ijuma committed:
Bump jacoco to 0.8.5

ijuma commented:
retest this please

omkreddy reviewed (approved):
@ijuma Thanks for the PR. LGTM if tests pass.

ijuma committed:
Replace `AES/GCM/PKCS5Padding` with `AES/GCM/NoPadding`

The former is invalid and JDK 14 rejects it, see
https://bugs.openjdk.java.net/browse/JDK-8229043.

ijuma committed:
Skip streams archetype for Java 14 (same as Java 11)

mjsax commented on jenkins.sh:
```diff
@@ -29,12 +29,17 @@
+if [ $JAVA_HOME = "/home/jenkins/tools/java/latest14" ] ; then
+  echo "Skipping Kafka Streams archetype test for Java 14"
+  exit 0
+fi
```
LGTM

ijuma commented:
2 flaky tests, one in each job:
* org.apache.kafka.streams.integration.EOSUncleanShutdownIntegrationTest.shouldWorkWithUncleanShutdownWipeOutStateStore[exactly_once_beta]
* kafka.integration.MetricsDuringTopicCreationDeletionTest.testMetricsDuringTopicCreateDelete
